src/trans.rs

```diff
@@ -1034,7 +1091,7 @@ impl<'v, 'tcx: 'v> BinaryenFnCtxt<'v, 'tcx> {
 
     #[inline]
     fn type_size(&self, ty: Ty<'tcx>) -> usize {
-        let substs = self.tcx.mk_substs(Substs::empty());
+        let substs = Substs::empty(*self.tcx);
```

Comment:
Ah, you're right.  Somehow my eyes just glazed over the `&'a` part of those fields. For posterity: https://github.com/rust-lang/rust/blob/f2b672d5567c3b20542e845baaf8a9c47d9df907/src/librustc/ty/context.rs#L285